src/EditorFeatures/Core/Implementation/NavigationBar/NavigationBarController.cs

```diff
@@ -181,87 +198,81 @@ private void OnWorkspaceChanged(object? sender, WorkspaceChangeEventArgs args)
         }
 
         private void OnViewFocused(object? sender, EventArgs e)
         {
             AssertIsForeground();
-            StartSelectedItemUpdateTask(delay: TaggerConstants.ShortDelay, updateUIWhenDone: true);
+            StartSelectedItemUpdateTask(delay: TaggerConstants.ShortDelay);
         }
 
         private void OnDropDownFocused(object? sender, EventArgs e)
         {
             AssertIsForeground();
 
-            // Refresh the drop downs to their full information
-            _waitIndicator.Wait(
-                EditorFeaturesResources.Navigation_Bars,
-                EditorFeaturesResources.Refreshing_navigation_bars,
-                allowCancel: true,
-                action: context => UpdateDropDownsSynchronously(context.CancellationToken));
-        }
-
-        private void UpdateDropDownsSynchronously(CancellationToken cancellationToken)
-        {
-            AssertIsForeground();
-
-            // If the presenter already has the full list and the model is already complete, then we
-            // don't have to do any further computation nor push anything to the presenter
-            if (PresenterAlreadyHaveUpToDateFullList(cancellationToken))
-            {
+            var document = _subjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+            if (document == null)
                 return;
-            }
-
-            // We need to ensure that all the state computation is up to date, so cancel any
-            // previous work and ensure the model is up to date
-            StartModelUpdateAndSelectedItemUpdateTasks(modelUpdateDelay: 0, selectedItemUpdateDelay: 0, updateUIWhenDone: false);
-
-            // Wait for the work to be complete. We'll wait with our cancellationToken, so if the
-            // user hits cancel we won't block them, but the computation can still continue
 
-            using (Logger.LogBlock(FunctionId.NavigationBar_UpdateDropDownsSynchronously_WaitForModel, cancellationToken))
-            {
-                _modelTask.Wait(cancellationToken);
-            }
+            // Grab and present whatever information we have at this point.
+            GetProjectItems(out var projectItems, out var selectedProjectItem);
+            var (model, selectedInfo) = _latestModelAndSelectedInfo_OnlyAccessOnUIThread;
 
-            using (Logger.LogBlock(FunctionId.NavigationBar_UpdateDropDownsSynchronously_WaitForSelectedItemInfo, cancellationToken))
+            if (Equals(model, _lastPresentedInfo.model) &&
+                Equals(selectedInfo, _lastPresentedInfo.selectedInfo) &&
+                Equals(selectedProjectItem, _lastPresentedInfo.selectedProjectItem) &&
+                projectItems.SequenceEqual(_lastPresentedInfo.projectItems))
             {
-                _selectedItemInfoTask.Wait(cancellationToken);
+                // Nothing changed, so we can skip presenting these items.
+                return;
             }
```

Comment:
this optimization is something we tested for in our tests.  i'm not really sure how useful it is, but i preserved it just in case.  I'm not sure why it would be a major problem to just open/close/open the dropdown and have us compute these items on each open.  However, in cae it is, we add this fast bail out.